tensorboard/backend/security_validator_test.py

```diff
@@ -0,0 +1,193 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Tests for `tensorboard.backend.security_validator`."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+try:
+  # python version >= 3.3
+  from unittest import mock    # pylint: disable=g-import-not-at-top
+except ImportError:
+  import mock    # pylint: disable=g-import-not-at-top,unused-import
+
+import werkzeug
+from werkzeug import test as werkzeug_test
+from werkzeug.datastructures import Headers
+
+from tensorboard import test as tb_test
+from tensorboard.backend import security_validator
+from tensorboard.util import tb_logging
+
+logger = tb_logging.get_logger()
+
+_WARN_PREFIX = "In 3.0, this warning will become an error:\n"
+
+
+def create_headers(
+    content_type="application/json",
+    x_content_type_options="nosniff",
+    content_secrurity_policy=""
+):
+  return Headers(
+      {
+          "Content-Type": content_type,
+          "X-Content-Type-Options": x_content_type_options,
+          "Content-Security-Policy": content_secrurity_policy,
+      }
+  )
+
+
+class SecurityValidatorMiddlewareTest(tb_test.TestCase):
+  """Tests for `SecurityValidatorMiddleware`."""
+
+  def setUp(self):
+    super(SecurityValidatorMiddlewareTest, self).setUp()
+    self.get_header = lambda req: {}
```

Comment:
Does this need to be a function?  It looks like it never actually uses the `req` argument so it could just be a property on the test case that we set.

Also optional but IMO a bit nicer if we can avoid the dependency on manipulating test case state indirectly to change the response.  What if we just passed in the CSP policy to return as a request header?  We'd implement the app request handler to just reflect that value back out, and then `make_request_and_maybe_assert_warn()` would just pass the headers to the call to `server.get()`.

Reply:
> What if we just passed in the CSP policy to return as a request header

I tried it and it was surprisingly hard because (1) content-type is auto inferred so reflection requires logic and (2) duplicative header name is omitted on `EnvironHeaders` making it more difficult to test the case. 

I took a stab at improving it.

Reply:
Sure, LGTM as is.